test/parallel/test-node-run.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+const common = require('../common');
+const { it, describe } = require('node:test');
+const assert = require('node:assert');
+
+const fixtures = require('../common/fixtures');
+const isLinuxOrOSX = common.isLinux || common.isOSX;
+
+describe('node run [command]', () => {
+  it('returns error on non-existent command', async () => {
+    const child = await common.spawnPromisified(
+      process.execPath,
+      [ 'run', 'test'],
+      { cwd: __dirname },
+    );
+    assert.match(child.stdout, /Can't read package\.json/);
+    assert.strictEqual(child.stderr, '');
+    assert.strictEqual(child.code, 1);
+  });
+
+  it('runs a valid command', { skip: !isLinuxOrOSX }, async () => {
+    // Run a script that just log `no test specified`
+    const child = await common.spawnPromisified(
+      process.execPath,
+      [ 'run', 'test'],
+      { cwd: fixtures.path('run-script') },
+    );
+    assert.match(child.stdout, /Error: no test specified/);
+    assert.strictEqual(child.code, 1);
+  });
+
+  it('adds node_modules/.bin to path', { skip: !isLinuxOrOSX }, async () => {
+    const child = await common.spawnPromisified(
+      process.execPath,
+      [ 'run', 'fastify'],
+      { cwd: fixtures.path('run-script') },
+    );
+    assert.match(child.stdout, /this is fastify/);
+    assert.strictEqual(child.stderr, '');
+    assert.strictEqual(child.code, 0);
+  });
+
+  it('appends positional arguments', { skip: !isLinuxOrOSX }, async () => {
+    const child = await common.spawnPromisified(
+      process.execPath,
+      [ 'run', 'positional-args', '--help'],
```

Comment:
We should have a test where an argument contains spaces, like `node run test "hello world test"` and `"hello world test"` is passed as a single argument.

Reply:
I replaced the test but there is an issue with echo not printing quotation marks, that I don't have time to investigate right now. Ref: https://unix.stackexchange.com/questions/24647/why-is-echo-ignoring-my-quote-characters

We need to land this in the next day or two if we want to include this in Node v22. So I recommend fixing these things later.

Reply:
Doing this work portably (windows + macos/linux) gets tricky.